### DISCUSSION TITLE
How to register unload hooks?

### DISCUSSION THREAD

**Discussion by binarykitchen** [storybookjs/storybook#29284] [Help]
### Summary

When switching stories, I am unable to abort any running processes from the previous story.

I'm maintaining a UI library using getUserMedia for video recording.

Imagine a story A where the video is being recorded, and then you switch to another story B, but story A still keeps recording.

A solution would be to add some clean up function or unload hook to stop the video recording before changing to the new iFrame with the new story.

Already researched but still haven't found a clue nor a solution yet.

How to solve this?

**Comment by valentinpalkovic** [comment]
Hi @binarykitchen 

Usually, if you use React, React's unmount method is called if you switch to another Story. Unfortunately, I don't quite understand why your component doesn't get destroyed.

**Comment by binarykitchen** [comment]
I'm not using React. Just good old plain JavaScript. 

So, just using the plain HTML renderer:
https://github.com/storybookjs/storybook/tree/next/code/renderers/html

And it doesn't come with unmount hooks. It would be nice to have.

**Reply by valentinpalkovic** [reply]
I see! Would you be interested to contribute and implement an unload hook?

**Reply by binarykitchen** [reply]
Not on Xmas Day :)

But yes, merry Xmas to you all for the wonderful work so far x
